internal/services/authorizer/authorizer_test.go

```diff
@@ -0,0 +1,79 @@
+package authorizer_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/services/authorizer"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+)
+
+func TestNew(t *testing.T) {
+	t.Parallel()
+
+	a := authorizer.New()
+
+	require.NotNil(t, a, "New authorizer is created")
+}
+
+func TestIsRequestFromRoot(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		currentUserNotRoot bool
+		noPeerCredsInfo    bool
+		noAuthInfo         bool
+
+		wantErr bool
+	}{
+		"Granted if current user considered as root": {},
+
+		"Error as deny when current user is not root": {currentUserNotRoot: true, wantErr: true},
+		"Error as deny when missing peer creds Info":  {noPeerCredsInfo: true, wantErr: true},
+		"Error as deny when missing auth info creds":  {noAuthInfo: true, wantErr: true},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			// Setup peer creds info
+			ctx := context.Background()
+			if !tc.noPeerCredsInfo {
+				var authInfo credentials.AuthInfo
+				if !tc.noAuthInfo {
+					uid := authorizer.CurrentUserUID()
+					authInfo = authorizer.NewTestPeerCredsInfo(uid, int32(uid))
+				}
+				p := peer.Peer{
+					AuthInfo: authInfo,
+				}
+				ctx = peer.NewContext(ctx, &p)
+			}
+
+			var opts []authorizer.Option
+			if !tc.currentUserNotRoot {
+				opts = append(opts, authorizer.WithCurrentUserAsRoot())
+			}
```

Comment:
Why not taking `authorizer.Option` in the test case directly?

Reply:
I think it’s more expressive to have a description of the behaviour in the test case, than an optional (test only) input. If the input was the one passed to the method, yes, but here, in `services/nss`, `services/pam`, we will have tons of those use case and just having the service modifier makes more sense to me.

Also, if one day, we change that helper, we would need to change 20 places instead of 3-4 :)